lib/services/embeddings_test.go

```diff
@@ -0,0 +1,149 @@
+package services_test
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+)
+
+var testEmbedding = []float32{0.1, 0.2, 0.3}
+
+// MockEmbedder returns embeddings based on the sha256 hash function. Those
+// embeddings have no semantic meaning but ensure different embedded content
+// provides different embeddings.
+type MockEmbedder struct {
+}
+
+func (m MockEmbedder) ComputeEmbeddings(_ context.Context, input []string) ([][]float32, error) {
+	result := make([][]float32, len(input))
+	for i, text := range input {
+		hash := sha256.Sum256([]byte(text))
+		vector := make([]float32, len(hash))
+		for j, x := range hash {
+			vector[j] = 1 / float32(int(x)+1)
+		}
+		result[i] = vector
+	}
+	return result, nil
+}
+
+func TestNodeEmbeddingWatcher(t *testing.T) {
```

Comment:
This test tests a lot of stuff. Can we split it into a few smaller/simpler tests? Otherwise, if something is broken, you don't have clear information what exactly happened.